### PR TITLE
[ACS-5877][APPS-2159][APPS-2163][APPS-2160][APPS-2164] Migration from moment.js to date-fns

### PR DESCRIPTION
**Pull request by arohilaGL**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

The files were using moment.js library

**What is the new behaviour?**
The files are using date-fns library

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/src/lib/common/utils/date-fns-utils.ts [hunk 1]
```diff
@@ -151,4 +153,15 @@ export class DateFnsUtils {
...
     }
+
+    /**
+     * Validates a date string using the specified date format.
+     *
+     * @param value - Date string to be matched.
+     * @param dateFormat - The date format string to use for matching.
+     * @returns true if given date string correct against the given format else will return false.
+     */
+    static isMatch(value: string, dateFormat: string): boolean {
```
Comment (rbahirsheth) [inline lib/core/src/lib/common/utils/date-fns-utils.ts:164]: Missed the unit test case.

File: lib/core/src/lib/form/services/widget-visibility.service.ts [hunk 2]
```diff
@@ -17,10 +17,10 @@
 
 import { LogService } from '../../common/services/log.service';
 import { Injectable } from '@angular/core';
-import moment from 'moment';
 import { FormFieldModel, FormModel, TabModel, ContainerModel, FormOutcomeModel } from '../components/widgets/core';
 import { TaskProcessVariableModel } from '../models/task-process-variable.model';
 import { WidgetVisibilityModel, WidgetTypeEnum } from '../models/widget-visibility.model';
+import { DateFnsUtils } from '../../common/utils/date-fns-utils';
```
Comment (rbahirsheth) [inline lib/core/src/lib/form/services/widget-visibility.service.ts:23]: I think having the path till common directive will work.

File: lib/core/src/lib/common/utils/date-fns-utils.ts (hunk 1 above)
Comment (pkunduGL) [reply lib/core/src/lib/common/utils/date-fns-utils.ts:164]: Added the unit test case

File: lib/core/src/lib/form/services/widget-visibility.service.ts (hunk 2 above)
Comment (pkunduGL) [reply lib/core/src/lib/form/services/widget-visibility.service.ts]: Fixed the path

File: lib/core/src/lib/common/utils/date-fns-utils.ts [hunk 3]
```diff
@@ -151,4 +153,15 @@ export class DateFnsUtils {
...
+        return isMatch(value, this.convertMomentToDateFnsFormat(dateFormat));
```
Comment (rbahirsheth) [inline lib/core/src/lib/common/utils/date-fns-utils.ts:165]: I think we don't need to use this method here 'this.convertMomentToDateFnsFormat(dateFormat)'.
simply pass the format like this
return isMatch(value, dateFormat);
Update the test case as per the change.

Comment (arohilaGL) [reply lib/core/src/lib/common/utils/date-fns-utils.ts:165]: The format, to my understanding, should change. Why would you suggest to not change the format?

File: lib/core/src/lib/common/utils/date-fns-utils.spec.ts [hunk 4]
```diff
@@ -55,4 +55,12 @@ describe('Date Format Translations', () => {
 
         expect(result).toEqual(expectedParsedDate);
     });
+    it('should match a date correctly', () => {
```
Comment (rbahirsheth) [inline lib/core/src/lib/common/utils/date-fns-utils.spec.ts:58]: please use new line for new test case.

File: lib/core/src/lib/common/utils/date-fns-utils.ts (hunk 3 above)
Comment (rbahirsheth) [reply lib/core/src/lib/common/utils/date-fns-utils.ts:165]: Got it. Thanks.

File: lib/core/src/lib/common/utils/date-fns-utils.spec.ts (hunk 4 above)
Comment (arohilaGL) [reply lib/core/src/lib/common/utils/date-fns-utils.spec.ts:59]: done

**Review by rbahirsheth — APPROVED** [review]

File: lib/core/src/lib/common/utils/date-fns-utils.spec.ts [hunk 5]
```diff
@@ -55,4 +55,13 @@ describe('Date Format Translations', () => {
 
         expect(result).toEqual(expectedParsedDate);
     });
+
```
Comment (AleksanderSklorz) [inline lib/core/src/lib/common/utils/date-fns-utils.spec.ts:58]: Please resolve lint issue: 
image

**Review by DenysVuika — APPROVED** [review]

File: lib/core/src/lib/common/utils/date-fns-utils.spec.ts (hunk 5 above)
Comment (arohilaGL) [reply lib/core/src/lib/common/utils/date-fns-utils.spec.ts]: done

**Review by AleksanderSklorz — APPROVED** [review]

**Review by AleksanderSklorz — APPROVED** [review]

File: lib/core/src/lib/form/services/widget-visibility.service.ts [hunk 6]
```diff
@@ -128,7 +128,7 @@ export class WidgetVisibilityService {
         } else if (visibilityObj.rightType === WidgetTypeEnum.field) {
             valueFound = this.getFormValue(form, visibilityObj.rightValue);
         } else {
-            if (moment(visibilityObj.rightValue, 'YYYY-MM-DD', true).isValid()) {
```
Comment (DenysVuika) [inline lib/core/src/lib/form/services/widget-visibility.service.ts:131]: marked PR as do not merge as this line needs tripple-checking that it works as expected with real data

**Pull request closed without merging**